tests/testthat/test_fetchingOmnipathWebserver.R

```diff
@@ -0,0 +1,145 @@
+library(OmnipathR)
+library(dplyr)
+
+################################################################################
+## Test of the functions fetching Omnipath Webserver
+################################################################################
+
+################################################################################
+## Test of the functions getting the list of Omnipath available databases
+################################################################################
+
+## .get_ptms_databases
+url_ptms <- 
+    'http://omnipathdb.org/ptms/?fields=sources&fields=references&genesymbols=1'
+ptms <- read.table(url_ptms, sep = '\t', header = TRUE, 
+    stringsAsFactors = FALSE)
+ptms_databases <- 
+    unique(unlist(strsplit(x = as.character(ptms$sources),split = ";")))
+
+## .get_interaction_databases
+url_interactions <- paste0('http://omnipathdb.org/interactions?',
+    'datasets=omnipath,pathwayextra,kinaseextra,ligrecextra',
+    ',tfregulons,mirnatarget&fields=sources,references&genesymbols=1')
+interactions <- read.table(url_interactions, sep = '\t', header = TRUE, 
+    stringsAsFactors = FALSE)
+interaction_databases  <- 
+    unique(unlist(strsplit(x = as.character(interactions$sources),split = ";")))
+
+## .get_complexes_databases
+url_complexes <- 'http://omnipathdb.org/complexes?&fields=sources'
+complexes <- read.csv(url_complexes, sep = '\t', header = TRUE,
+    stringsAsFactors = FALSE)
+complexes_databases <-
+    unique(unlist(strsplit(x = as.character(complexes$sources),split = ";")))
+
+## .get_annotation_databases
+annotations_db <- 
+    c("Adhesome","SignaLink3","TopDB","HGNC","Ramilowski_location", 
```

Comment:
I think the list of resources should not be hardcoded here, instead you can use the http://omnipathdb.org/annotations_summary query and unique the first column. Similarly for the intercell.

Reply:
Yeah, you are right. Indeed, I wanted to ask you how to get the different sources (databases) in a more efficient way from the webservice. For **interactions**, **ptms**, **complexes** and **intercell** (it is not hardcoded), we download all the records from the webservice and we get all the different databases from there (Not efficient at all). Too many records for **annotations**, so the query took long time and I decided to hardcode it. Is there any way to obtain from the websersice also a "summary" for the rest types of queries?

Reply:
I cloned this branch to take a look at the R package and I realized this issue yesterday evening. So I added this functionality to the pypath web server, but on the public web service haven't updated yet. Will do it very soon.